src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -142,3 +142,23 @@ QgsValueRelationFieldFormatter::ValueRelationCache QgsValueRelationFieldFormatte
 
   return cache;
 }
+
+QStringList QgsValueRelationFieldFormatter::valueToStringList( const QVariant &value )
+{
+  QStringList checkList;
+  if ( value.type() == QVariant::StringList )
+    checkList = value.toStringList();
+  else if ( value.type() == QVariant::String )
+    checkList = value.toString().remove( QChar( '{' ) ).remove( QChar( '}' ) ).split( ',' );
+  else if ( value.type() == QVariant::List )
+  {
+    QVariantList valuesList( value.toList( ) );
+    for ( const QVariant &listItem : qgis::as_const( valuesList ) )
```

Comment:
Better make the list itself const (i.e. remove `qgis::as_const` here and add a `const` one line up)